File: src/main/kotlin/kafka/rest/admin/domain/TopicService.java
package kafka.rest.admin.domain;

import org.apache.kafka.clients.consumer.Consumer;
import org.apache.kafka.common.PartitionInfo;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Map;
import java.util.Set;
import java.util.stream.Collectors;

@Service
public class TopicService {
	public Set<String> list(KafkaConsumerFactory kafkaConsumerFactory) {
		try (Consumer<Object, Object> consumer = kafkaConsumerFactory.createConsumer()) {
			Map<String, List<PartitionInfo>> map = consumer.listTopics();
			return map.keySet();
		}
	}

	public List<String> get(KafkaConsumerFactory kafkaConsumerFactory, String name) {
		try (Consumer<Object, Object> consumer = kafkaConsumerFactory.createConsumer()) {
			Map<String, List<PartitionInfo>> map = consumer.listTopics();
			return map.get(name).stream().map(PartitionInfo::toString).collect(Collectors.toList());
		}
	}
}

File: src/main/kotlin/kafka/rest/admin/domain/TopicService.kt
package kafka.rest.admin.domain

import org.apache.kafka.common.PartitionInfo
import org.springframework.stereotype.Service
import java.util.stream.Collectors

@Service
class TopicService {
    fun list(kafkaConsumerFactory: KafkaConsumerFactory): Set<String> {
        kafkaConsumerFactory.createConsumer().use { consumer ->
            val map = consumer.listTopics()
            return map.keys
        }
    }

    fun get(kafkaConsumerFactory: KafkaConsumerFactory, name: String?): List<String> {
        kafkaConsumerFactory.createConsumer().use { consumer ->
            val map = consumer.listTopics()
            return map[name]!!.stream().map { obj: PartitionInfo -> obj.toString() }.collect(Collectors.toList())
        }
    }
}